lib/neovim-plugin.nix

```diff
@@ -111,7 +111,7 @@ with lib;
             extraPlugins = (optional installPackage cfg.package) ++ extraPlugins;
             inherit extraPackages;
 
-            ${extraConfigNamespace} = optionalString callSetup ''
+            ${extraConfigNamespace} = mkIf callSetup ''
```

Comment:
This is intentionally not using mkIf

Reply:
Why? If `callSetup` is false, plugin produces empty an string value. Later on all this empty strings are concatenated with newlines. If it can be fixed in other places, why not here?

Reply:
As a general rule, these general use helpers don't know _why_ a feature such as `callSetup` has been disabled, it may be because the caller doesn't want something that feature uses to be evaluated (which mkIf won't solve).

To clarify, `mkIf` still evaluates the content, even when the condition is false.

In this specific case, we'd be calling `toLuaObject cfg.settings` unnecessarily, although maybe that's not a big deal.

But here I think it is better to use `optional`, `optionals`, `optionalAttrs`, `optionalString` or `if then else` as appropriate.

Reply:
> To clarify, mkIf still evaluates the content, even when the condition is false.

OK. Got it. Since `mkMerge` is already used here, what do you thin about:

```nix
(optionalAttrs callSetup {
   ${extraConfigNamespace} = ''
       require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
    '';
})
```

I think it preserve current behavior and avoid empty string.

Reply:
Agreed, I think that's the right approach 😃